SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -401,14 +394,27 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		return;
 	}
 
-	// If we're the client, we can now ask the server to spawn our controller.
-	if (!IsServer())
+	if (IsServer())
+	{
+		if (GlobalStateManager && !ServerConnection)
+		{
+			GlobalStateManager->SetCanBeginPlay(true);
+			GlobalStateManager->TriggerBeginPlay();
+			GlobalStateManager->SetAcceptingPlayers(true);
+		}
+	}
+	else
 	{
+		// If we're the client, we can now ask the server to spawn our controller.
 		// If we know the GSM is already accepting players, simply spawn.
-		if (GlobalStateManager->bAcceptingPlayers && GetWorld()->RemovePIEPrefix(GlobalStateManager->DeploymentMapURL) == GetWorld()->RemovePIEPrefix(GetWorld()->URL.Map))
+		UWorld* World = GetWorld();
+		if (GlobalStateManager->GetAcceptingPlayers() &&
```

Comment:
I don't think it's guaranteed that the component updates for the GSM in Spatial will come down off the wire at the same time. Therefore I think it's possible that we could get AcceptingPlayers = true but with an old SessionID (component update not processed yet) and thus we would hit the else statement here and crash no? 

Shouldn't we be waiting for the SessionID in the GSM to match the SessionID we have from the URL instead?

Reply:
✓